test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs

```diff
@@ -49,99 +46,326 @@ public void ToOtlpResourceMetricsTest(bool includeServiceNameInResource)
-
-            var exportedItems = new List<Metric>();
-
             using var provider = Sdk.CreateMeterProviderBuilder()
                 .SetResourceBuilder(resourceBuilder)
                 .AddMeter(meter.Name)
-                .AddReader(new BaseExportingMetricReader(new InMemoryExporter<Metric>(exportedItems))
-                {
-                    Temporality = AggregationTemporality.Delta,
-                })
+                .AddInMemoryExporter(metrics)
                 .Build();
 
             var counter = meter.CreateCounter<int>("counter");
+            counter.Add(100);
+
+            provider.ForceFlush();
+
+            var batch = new Batch<Metric>(metrics.ToArray(), metrics.Count);
+
+            var request = new OtlpCollector.ExportMetricsServiceRequest();
+            request.AddMetrics(resourceBuilder.Build().ToOtlpResource(), batch);
+
+            Assert.Single(request.ResourceMetrics);
+            var resourceMetric = request.ResourceMetrics.First();
+            var oltpResource = resourceMetric.Resource;
+
+            if (includeServiceNameInResource)
+            {
+                Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.StringValue == "service-name");
+                Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceNamespace && kvp.Value.StringValue == "ns1");
+            }
+            else
+            {
+                Assert.Contains(oltpResource.Attributes, (kvp) => kvp.Key == ResourceSemanticConventions.AttributeServiceName && kvp.Value.ToString().Contains("unknown_service:"));
+            }
 
-            counter.Add(100, tags);
+            Assert.Single(resourceMetric.InstrumentationLibraryMetrics);
+            var instrumentationLibraryMetrics = resourceMetric.InstrumentationLibraryMetrics.First();
+            Assert.Equal(string.Empty, instrumentationLibraryMetrics.SchemaUrl);
+            Assert.Equal(meter.Name, instrumentationLibraryMetrics.InstrumentationLibrary.Name);
+            Assert.Equal("0.0.1", instrumentationLibraryMetrics.InstrumentationLibrary.Version);
+        }
+
+        [Theory]
+        [InlineData("test_gauge", null, null, 123, null)]
+        [InlineData("test_gauge", null, null, null, 123.45)]
+        [InlineData("test_gauge", "description", "unit", 123, null)]
+        public void TestGaugeToOtlpMetric(string name, string description, string unit, long? longValue, double? doubleValue, params object[] keysValues)
+        {
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
 
-            var testCompleted = false;
+            var attributes = ToAttributes(keysValues).ToArray();
```

Comment:
doesnt seem used.

Reply:
Yep you're right. I removed it . No way to set attributes using ObservableGauge right?

Reply:
Observable can also have attributes! See this example
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/learning-more-instruments/Program.cs#L56

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/794450b17808738a9f66543691a8cfaab1d83d9b/examples/Console/TestPrometheusExporter.cs#L63